tuned_lens/nn/lenses.py

```diff
@@ -316,3 +316,66 @@ def forward(self, h: th.Tensor, idx: int) -> th.Tensor:
     def __len__(self) -> int:
         """Return the number of layer translators in the lens."""
         return len(self.layer_translators)
+
+    @th.inference_mode()
+    def generate(
+        self,
+        model: PreTrainedModel,
+        layer: int,
+        input_ids: th.Tensor,
+        do_sample: bool = True,
+        temp: float = 1.0,
+        max_new_tokens: int = 100,
+    ) -> th.Tensor:
+        """Generate from the tuned lens at the given layer.
+
+        Args:
+            model: The base model the generate from. Usually the model this lens trained
+                on.
+            layer: The layer to generate from.
+            input_ids: (batch x prompt_len) The input ids to generate from.
+            do_sample: Whether to use sampling or greedy decoding.
+            temp: The temperature to use for sampling.
+            max_new_tokens: The maximum number of tokens to generate.
+
+        Returns:
+            The prompt concatenated with the newly generated tokens.
+        """
+        eos_token = model.generation_config.eos_token_id
+
+        tokens = input_ids.clone()
```

Comment:
There doesn't seem to be any need to clone this ? Maybe then you'd have to add to the smoke test "the input is unmodified".